test/e2e/set/scyllacluster/scyllacluster_expose.go

```diff
@@ -4,26 +4,258 @@ package scyllacluster
 
 	f := framework.NewFramework("scyllacluster")
 
+	type entry struct {
+		exposeOptions            *scyllav1.ExposeOptions
+		memberServiceCreatedHook func(ctx context.Context, client corev1client.CoreV1Interface, svc *corev1.Service)
+		validateService          func(svc *corev1.Service)
+		validateScyllaConfig     func(ctx context.Context, configClient *scyllaclient.ConfigClient, svc *corev1.Service, pod *corev1.Pod)
+	}
+
+	g.DescribeTable("should be exposed", func(e *entry) {
+		ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+		defer cancel()
+
+		sc := scyllafixture.BasicScyllaCluster.ReadOrFail()
+		sc.Spec.Datacenter.Racks[0].Members = 1
+		sc.Spec.ExposeOptions = e.exposeOptions
+
+		framework.By("Creating a ScyllaCluster")
+		sc, err := f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Create(ctx, sc, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for member Service to be created")
+		waitCtx, waitCtxCancel := utils.ContextForRollout(ctx, sc)
+		defer waitCtxCancel()
+
+		svcName := naming.MemberServiceName(sc.Spec.Datacenter.Racks[0], sc, 0)
+		svc, err := utils.WaitForServiceState(waitCtx, f.KubeClient().CoreV1().Services(sc.Namespace), svcName, utils.WaitForStateOptions{}, func(svc *corev1.Service) (bool, error) {
+			return svc != nil, nil
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		if e.memberServiceCreatedHook != nil {
+			e.memberServiceCreatedHook(ctx, f.KubeAdminClient().CoreV1(), svc)
+		}
+
+		framework.By("Waiting for ScyllaCluster to deploy")
+
+		sc, err = utils.WaitForScyllaClusterState(waitCtx, f.ScyllaClient().ScyllaV1(), sc.Namespace, sc.Name, utils.WaitForStateOptions{}, utils.IsScyllaClusterRolledOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		verifyScyllaCluster(ctx, f.KubeClient(), sc)
+
+		svc, err = f.KubeClient().CoreV1().Services(sc.Namespace).Get(ctx, svcName, metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		e.validateService(svc)
+
+		pod, err := f.KubeClient().CoreV1().Pods(sc.Namespace).Get(ctx, svcName, metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		configClient, err := utils.GetScyllaConfigClient(ctx, f.KubeClient().CoreV1(), sc, pod.Status.PodIP)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		e.validateScyllaConfig(ctx, configClient, svc, pod)
+
+		if utilfeature.DefaultMutableFeatureGate.Enabled(features.AutomaticTLSCertificates) {
+			serviceAndPodIPs, err := utils.GetNodesServiceAndPodIPs(ctx, f.KubeClient().CoreV1(), sc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			hostsIPs, err := helpers.ParseClusterIPs(serviceAndPodIPs)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			var servingDNSNames []string
+			services, err := f.KubeClient().CoreV1().Services(sc.Namespace).List(ctx, metav1.ListOptions{
+				LabelSelector: labels.SelectorFromSet(naming.ClusterLabels(sc)).String(),
+			})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			for _, svc := range services.Items {
```

Comment:
pls assert the length so this silently doesn't pass with 0 or less svcs then we expect

Reply:
done